snippets/puppet_setup.erb

```diff
@@ -58,4 +58,26 @@ fi
 <% end -%>
 <% end -%>
 
-<%= bin_path %>/puppet agent --config <%= etc_path %>/puppet.conf --onetime --tags no_such_tag <%= @host.puppetmaster.blank? ? '' : "--server #{@host.puppetmaster}" %> --no-daemonize
+<% if @host.param_true?('run-puppet-in-installer') -%>
+<% if @host.operatingsystem.name == 'Ubuntu' and @host.operatingsystem.major.to_i >= 15 -%>
+# Puppet tries to detect the init system in use by checking for the presence 
+# of the directory /run/systemd/system. That detection fails in a chroot 
+# environment like the one the installer provides.
+#
+# Related puppet tickets:
+# https://tickets.puppetlabs.com/browse/PA-136
+# https://tickets.puppetlabs.com/browse/PUP-5577
+#
+# We work around that here until it gets fixed in Puppet (maybe never for 
+# Puppet 3.x?)
+mkdir -p /run/systemd/system
+<% end -%>
+<% end -%>
```

Comment:
It's executed if puppet-run-in-installer == true && ubuntu > 15, and before we actually run puppet. That's intended. Or are we talking past each other?